File: patterns/c-patterns/pattern30.c
// Pattern 30: ...

// * A* B*
// C* D* E
// * F* G*
// H* I* J
// * K* L*

#include <stdio.h>
int main()
{
    int n = 5;
    int i, j;

    char x = 'A';

    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= n; j++)
        {
            if ((i + j) % 2 == 1)
            {
                printf("%c", x++);
            }
            else
            {
                printf("* ");
            }
        }
        printf("\n");
    }
    return 0;
}

File: patterns/python-patterns/pattern30.py
# Pattern 30: ...

# * A* B*
# C* D* E
# * F* G*
# H* I* J
# * K* L*

n = 5
x = ord('A')

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if (i + j) % 2 == 1:
            print(chr(x), end="")
            x += 1
        else:
            print("* ", end="")
    print()
